pkg/types/intoto/v0.0.2/entry.go

```diff
@@ -0,0 +1,418 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package intoto
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/in-toto/in-toto-golang/in_toto"
+	"github.com/secure-systems-lab/go-securesystemslib/dsse"
+	"github.com/spf13/viper"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sigstore/rekor/pkg/log"
+	"github.com/sigstore/rekor/pkg/pki/x509"
+	"github.com/sigstore/rekor/pkg/types"
+	"github.com/sigstore/rekor/pkg/types/intoto"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature/options"
+)
+
+const (
+	APIVERSION = "0.0.2"
+)
+
+func init() {
+	if err := intoto.VersionMap.SetEntryFactory(APIVERSION, NewEntry); err != nil {
+		log.Logger.Panic(err)
+	}
+}
+
+type V002Entry struct {
+	IntotoObj models.IntotoV002Schema
+	env       dsse.Envelope
+}
+
+func (v V002Entry) APIVersion() string {
+	return APIVERSION
+}
+
+func NewEntry() types.EntryImpl {
+	return &V002Entry{}
+}
+
+func (v V002Entry) IndexKeys() ([]string, error) {
+	var result []string
+
+	if v.IntotoObj.Content == nil || v.IntotoObj.Content.Envelope == nil {
+		log.Logger.Info("IntotoObj content or dsse envelope is nil")
+		return result, nil
+	}
+
+	for _, sig := range v.IntotoObj.Content.Envelope.Signatures {
+		keyHash := sha256.Sum256(sig.PublicKey)
```

Comment:
maybe take the `keyObj` below's CanonicalValue for the index of the key?

Reply:
done.